src/include/binder/binder_context.h

```diff
@@ -14,6 +14,7 @@
 #include <string>
 
 #include "type/types.h"
+#include "concurrency/transaction.h"
```

Comment:
I agree with @pmenon. You should forward declare the `concurrency::Transaction` class.